### PR TITLE
Neosec Pack

### PR DESCRIPTION
**Pull request by gala-neosec**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Neosec/Integrations/Neosec/Neosec.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+            latest_created_time = int(incident_created_time)
+
+    # Save the next_run as a dict with the last_fetch key to be stored
+    next_run = {"last_fetch": latest_created_time}
+    return next_run, incidents
+
+
+def set_alert_status(client: NeosecClient, alert_id: str, alert_status: str) -> None:
+    client.patch_alert(alert_id, alert_status)
+
```
Comment (ostolero) [inline Packs/Neosec/Integrations/Neosec/Neosec.py:311]: return a proper response to the user to indicate if the action was successful. something like -
```
    res = client.patch_alert(alert_id, alert_status)
    if res.status == 200:
        return_results("Alert {alert_id} updated successfully")
    else:
        return_error(f"Error updating alert {alert_id} - {res.content}")
```

File: Packs/Neosec/pack_metadata.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "Neosec",
+    "description": "Utilize Neosec behavioral analytics to protect your API estate from OWSP top 10 vulnerabilities and suspicious user behavior.",
+    "support": "community",
```
Comment (ostolero) [inline Packs/Neosec/pack_metadata.json:4]: I see you registered as a partner, change this to 
`"support": "partner",`

**Review by ostolero — CHANGES_REQUESTED** [review]
Hi , everything looks really good! see my few changes.
I'm adding one of our security team member to go over the incident field, type, classifier and layout.
If you have any questions feel free to contact me here or through slack. tnx!

**Comment by efelmandar** [comment]
@gala-neosec Hi and thank you for your contribution, I've finished reviewing the pack's content and would like if we could schedule a demo to demonstrate the use of the integration and content, please let us know your available time slots? @ostolero FYI

File: Packs/Neosec/Integrations/Neosec/Neosec.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  required: false
+  defaultvalue: https://neograph.neosec.com
+  type: 0
+  additionalinfo: ""
+- name: tenant_key
+  display: Tenant Key
+  required: false
+  defaultvalue:
+  type: 0
+  additionalinfo: ""
```
Comment (ostolero) [inline Packs/Neosec/Integrations/Neosec/Neosec.yml:25]: Add the `required: true` field to both

File: Packs/Neosec/Integrations/Neosec/Neosec.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      predefined:
+      - ""
+    - name: alert_status
+      description: The alert status, the options are "Open" or "Closed".
+      required: true
+      isArray: false
+      defaultValue: ""
+      predefined:
+      - Open
+      - Closed
```
Comment (ostolero) [inline Packs/Neosec/Integrations/Neosec/Neosec.yml:112]: I see you had the intention to have this as a predefined argument, I think adding this field will fix the issue and show the options in XSOAR
`auto: PREDEFINED`

**Review by ostolero — CHANGES_REQUESTED** [review]
Hi @gala-neosec , Thank you for a great demo! learned a lot about your product and the integration seems to work perfectly.
Regarding the few points raised in the demo-
1. See my two comments in the yml file regarding mandatory params and multi value argument.
2.After looking into it, `proxy` and `insecure` support are a recommendation not a requirement, you can see an example in HelloWorld integration if you want to add them.
3. Add the extra fields to the mapper as we discussed @efelmandar will look an example of a grid mapper.
Thank you, let me know if you have any questions :)

**Comment by arsenyk-neosec** [comment]
> 

@efelmandar don't forget to send grid field example

**Comment by efelmandar** [comment]
@arsenyk-neosec I've sent you an email last week with the documentation and example of how to use grid field, did you get it? 
if you need additional info/help with this please let me know

**Comment by arsenyk-neosec** [comment]
> @arsenyk-neosec I've sent you an email last week with the documentation and example of how to use grid field, did you get it? if you need additional info/help with this please let me know

Thanks for the example. @gala-neosec haven't got to it yet. we will reach out when we will finish the fixes

**Review by ostolero — APPROVED** [review]
Hi @gala-neosec , changes look great! will just ask @efelmandar to have a final look and we can merge :)

**Comment by arsenyk-neosec** [comment]
> @gala-neosec I went over the changes and it looks excellent and we're ready to merge, @ostolero FYI

@ostolero what are the next steps?

**Pull request merged**